commons-numbers-complex/src/main/java/org/apache/commons/numbers/complex/ComplexFunctions.java

```diff
@@ -396,6 +399,255 @@ public static <R> R proj(double real, double imaginary, ComplexSink<R> action) {
         return action.apply(real, imaginary);
     }
 
+    /**
+     * Returns a {@code Object} whose value is {@code (real1 + real2, imaginary1 + imaginary2)}.
+     * Implements the formula:
+     *
+     * <p>\[ (a + i b) + (c + i d) = (a + c) + i (b + d) \]
+     *
+     * @param re1 Real part \( a \) of the first complex number \( (a +ib) \).
+     * @param im1 Imaginary part \( b \) of the first complex number \( (a +ib) \).
+     * @param re2 Real part \( a \) of the second complex number \( (a +ib) \).
+     * @param im2 Imaginary part \( b \) of the second complex number \( (a +ib) \).
+     * @param action Consumer for the final added complex number.
+     * @param <R> the return type of the supplied action.
+     * @return the object returned by the supplied action.
+     * @see <a href="http://mathworld.wolfram.com/ComplexAddition.html">Complex Addition</a>
+     */
+    public static <R> R add(double re1, double im1, double re2, double im2, ComplexSink<R> action) {
```

Comment:
It may be very verbose but for consistency we should use `real1/2` and `imaginary1/2` as these are the names used for all the unary functions. This matches what you have put in the javadoc too.